### PR TITLE
Setup for Reports and Admin Banner Stores

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Creating the stores, initial states, and actions for reports and admin banners, as well as a global store.

### Related ticket(s)

---
### How to test

This one is a bit difficult to test, mostly it'll be for the `userStore` at this point, but once plug in all of the API calls and whatnot, the result should be a bit clearer.

If you log in as any user and use the Redux DevTools extension, you should see something like this:

Screenshot 2023-08-30 at 5 24 56 PM

The global state of the current user you're logged in as!

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by ajaitasaini — APPROVED** [review]
looks good to me 😃

File: services/ui-src/src/components/reports/ReportProvider.tsx [hunk 1]
```diff
@@ -24,3 +28,145 @@ export const ReportContext = createContext<ReportContextShape>({
...
+    () => ({
+      // report
+      report,
+      contextIsLoaded,
+      createReport,
+      // reports by state
+      reportsByState,
+      fetchReportsByState,
+      // selected report
+      clearReportSelection,
```
Comment (britt-mo) [inline services/ui-src/src/components/reports/ReportProvider.tsx:146]: are these commented out items going to be used? Just want to confirm.

**Review by britt-mo — DISMISSED** [review]

File: services/ui-src/src/components/app/App.tsx [hunk 2]
```diff
@@ -38,21 +40,23 @@ export const App = () => {
...
-          <SkipNav
-            id="skip-nav-main"
-            href={"#main-content"}
-            text={`Skip to ${"main content"}`}
-            sxOverride={sx.skipnav}
-          />
+          <ReportProvider>
+            <Timeout />
+            <MainSkipNav />
+            {/* id="skip-nav-main"
```
Comment (cassandradanger) [inline services/ui-src/src/components/app/App.tsx:46]: do we need this commented out code?

File: services/ui-src/src/components/modals/AddEditReportModal.tsx [hunk 3]
```diff
@@ -91,51 +90,32 @@ export const AddEditReportModal = ({
...
+        ...dataToWrite.fieldData,
+        stateName: States[activeState as keyof typeof States],
+        submissionCount: reportType === "WP" ? 0 : undefined,
+        // All new WP reports are NOT resubmissions by definition.
+        versionControl:
+          reportType === "WP"
+            ? [
+                {
+                  // pragma: allowlist nextline secret
+                  key: "versionControl-KFCd3rfEu3eT4UFskUhDtx",
```
Comment (cassandradanger) [inline services/ui-src/src/components/modals/AddEditReportModal.tsx:111]: is this something we should have showing on the frontend?

**Review by BearHanded — APPROVED** [review]

**Review by cassandradanger — APPROVED** [review]

**Pull request merged**